core/trino-main/src/test/java/io/trino/execution/TestSqlTaskManager.java

```diff
@@ -463,19 +504,28 @@ public ListenableFuture<Void> processFor(Duration duration)
                     break;
                 }
             }
-            interrupted.set(true);
+
+            finishedFuture.set(null);
```

Comment:
Move it after assigning `closed`, so once the `finishedFuture` is set the `isFinished` method returns `true`